File: sistema-evaluaciones-frontend/src/app/services/examen.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import baseUrl from './helper';

@Injectable({
  providedIn: 'root'
})
export class ExamenService {

  constructor(private http:HttpClient) { }

  public listarCuestionarios(){
    return this.http.get(`${baseUrl}/examen/`);
  }

  public agregarCuestionario(examen:any){
    return this.http.post(`${baseUrl}/examen/`,examen);
  
  }

  public eliminarExamen(examenId:any){
    return this.http.delete(`${baseUrl}/examen/${examenId}`);
  }

  public obtenerCuestionario(examenId:any){
    return this.http.get(`${baseUrl}/examen/${examenId}`);
  }

  public actualizarCuestionario(examen:any){
    return this.http.put(`${baseUrl}/examen/`,examen);
  
  }

  public listarExamenesDeUnaCategoria(categoriaId:any){
    return this.http.get(`${baseUrl}/examen/categoria/${categoriaId}`);
  }

  public obtenerExamenesActivos(){
    return this.http.get(`${baseUrl}/examen/activo`);
  }

  public obtenerExamenesActivosDeUnaCategoria(categoriaId:any){
    return this.http.get(`${baseUrl}/examen/categoria/activo/${categoriaId}`);
  }

}

File: sistema-examenes-backend/src/main/java/com/sistemaexamenes/sistemaexamenesbackend/servicios/ExamenService.java
package com.sistemaexamenes.sistemaexamenesbackend.servicios;

import com.sistemaexamenes.sistemaexamenesbackend.entidades.Categoria;
import com.sistemaexamenes.sistemaexamenesbackend.entidades.Examen;

import java.util.List;
import java.util.Set;

public interface ExamenService {

    Examen agregarExamen(Examen examen);

    Examen actualizarExamen(Examen examen);

    Set<Examen> obtenerExamenes();

    Examen obtenerExamen(Long examenId);

    void eliminarExamen(Long examenId);


    List<Examen> listarExamenesDeUnaCategoria(Categoria categoria);

    List<Examen> obtenerExamenesActivos();

    List<Examen> obtenerExamenesActivosDeUnaCategoria(Categoria categoria);


}
